### ISSUE TITLE
Specify allowed 'subtype' for studies/activities and check it

### ISSUE THREAD

**Issue by madprime** [PersonalGenomesOrg/open-humans#269]
In `common/models.py` the `BaseDataFile` has a `subtype` field. Allowed values for this should be specified for all data import apps (in studies or activities). Right now this appears to already be done for `activities/runkeeper` in `apps.py`. It should always be done.

When open-humans-data-processing reports a completed data retrieval task, it posts an update that describes new files, along with subtypes for each file. We should check that the subtype matches the allowed types and report an error/alert if this isn't true.

If no subtype is defined, a default might be assigned for apps that only have one possible subtype. (This currently occurs for go_viral and american_gut.) The check described above should occur after the default is applied (to check that the default assignment was consistent with the allowed values).

**Comment by beaugunderson** [comment]
this will depend on #291, which will add the `subtypes` field to the app.

**Comment by madprime** [comment]
Won't fix. We don't like subtypes anymore.
